QueueAdvancedTest.testTakeInterruption & testPutInterruption

```
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at com.hazelcast.collection.impl.queue.QueueAdvancedTest.testTakeInterruption(QueueAdvancedTest.java:601)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-OpenJDK6/672/testReport/

```
java.lang.AssertionError: Expected queue.take() to be interrupted, but interrupted state is: false
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.collection.impl.queue.QueueAdvancedTest.testTakeInterruption(QueueAdvancedTest.java:635)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-OracleJDK8/com.hazelcast$hazelcast/719/testReport/junit/com.hazelcast.collection.impl.queue/QueueAdvancedTest/testTakeInterruption/

```
java.lang.AssertionError: Expected queue.put() to be interrupted, but interrupted state is: false
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.collection.impl.queue.QueueAdvancedTest.testPutInterruption(QueueAdvancedTest.java:668)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x/com.hazelcast$hazelcast/4856/testReport/junit/com.hazelcast.collection.impl.queue/QueueAdvancedTest/testPutInterruption/

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x/4858/

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x/com.hazelcast$hazelcast/4861/testReport/junit/com.hazelcast.collection.impl.queue/QueueAdvancedTest/testPutInterruption/

@pveentjer It seems that we have failures again
```
java.lang.AssertionError: error instanceof com.hazelcast.core.OperationTimeoutException: OfferOperation invocation failed to complete due to operation-heartbeat-timeout. Current time: 2016-05-06 13:48:24.879. Total elapsed time: 2250 ms. Last heartbeat: never. Last member heartbeat: 2016-05-06 13:48:24.879. Invocation{op=com.hazelcast.collection.impl.queue.operations.OfferOperation{serviceName='hz:impl:queueService', identityHash=1322919554, partitionId=41, replicaIndex=0, callId=0, invocationTime=1462531702629 (2016-05-06 13:48:22.629), waitTimeout=-1001, callTimeout=1000}, tryCount=250, tryPauseMillis=500, invokeCount=0, callTimeoutMillis=1000, firstInvocationTimeMs=1462531702629, firstInvocationTime='2016-05-06 13:48:22.629', lastHeartbeatMillis=0, lastHeartbeatTime='1970-01-01 02:00:00.000', target=[127.0.0.1]:5570, pendingResponse={VOID}, backupsAcksExpected=0, backupsAcksReceived=0, connection=null}
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.test.TestThread.assertFailsEventually(TestThread.java:73)
	at com.hazelcast.collection.impl.queue.QueueAdvancedTest.testPutInterruption(QueueAdvancedTest.java:658)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x/5022/#showFailuresLink

Tests are failing again:
```
java.lang.AssertionError: Expected queue.put() to be interrupted, but interrupted state is: false
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.collection.impl.queue.QueueAdvancedTest.testPutInterruption(QueueAdvancedTest.java:668)
```
https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.maintenance-ZuluJDK6/lastCompletedBuild/com.hazelcast$hazelcast/testReport/com.hazelcast.collection.impl.queue/QueueAdvancedTest/testPutInterruption/

```
java.lang.AssertionError: Expected queue.take() to be interrupted, but interrupted state is: false
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.collection.impl.queue.QueueAdvancedTest.testTakeInterruption(QueueAdvancedTest.java:635)
```
https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.maintenance-ZuluJDK6/lastCompletedBuild/com.hazelcast$hazelcast/testReport/com.hazelcast.collection.impl.queue/QueueAdvancedTest/testTakeInterruption/

--- a/hazelcast/src/test/java/com/hazelcast/collection/impl/queue/QueueAdvancedTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/collection/impl/queue/QueueAdvancedTest.java
@@ -28,6 +28,7 @@ import com.hazelcast.instance.GroupProperty;
 import com.hazelcast.test.HazelcastParallelClassRunner;
 import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.test.TestHazelcastInstanceFactory;
+import com.hazelcast.test.TestThread;
 import com.hazelcast.test.annotation.ParallelTest;
 import com.hazelcast.test.annotation.QuickTest;
 import com.hazelcast.util.EmptyStatement;
@@ -41,7 +42,6 @@ import java.util.Queue;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
-import java.util.concurrent.atomic.AtomicBoolean;
 
 import static java.util.concurrent.TimeUnit.MINUTES;
 import static java.util.concurrent.TimeUnit.SECONDS;
@@ -612,27 +612,22 @@ public class QueueAdvancedTest extends HazelcastTestSupport {
 
         HazelcastInstance instance = createHazelcastInstance(config);
         final IQueue<Thread> queue = instance.getQueue(randomName());
-        final AtomicBoolean interrupted = new AtomicBoolean();
 
-        final CountDownLatch latch = new CountDownLatch(1);
-        Thread thread = new Thread() {
+        final CountDownLatch takeLatch = new CountDownLatch(1);
+        TestThread thread = new TestThread() {
             @Override
-            public void run() {
-                try {
-                    latch.countDown();
-                    queue.take();
-                } catch (InterruptedException e) {
-                    interrupted.set(true);
-                }
+            public void doRun() throws Throwable {
+                takeLatch.countDown();
+                queue.take();
             }
         };
         thread.start();
 
-        latch.await(10, SECONDS);
-        thread.interrupt();
-        thread.join(5000);
+        assertOpenEventually(takeLatch);
 
-        assertTrue("Expected queue.take() to be interrupted, but interrupted state is: " + interrupted, interrupted.get());
+        thread.interrupt();
+
+        thread.assertFailsEventually(InterruptedException.class);
     }
 
     @Test
@@ -643,29 +638,24 @@ public class QueueAdvancedTest extends HazelcastTestSupport {
 
         HazelcastInstance instance = createHazelcastInstance(config);
         final IQueue<String> queue = instance.getQueue(randomName());
-        final AtomicBoolean interrupted = new AtomicBoolean();
 
         assertTrue("Expected queue.offer() to succeed", queue.offer("item"));
 
-        final CountDownLatch latch = new CountDownLatch(1);
-        Thread thread = new Thread() {
+        final CountDownLatch putLatch = new CountDownLatch(1);
+        TestThread thread = new TestThread() {
             @Override
-            public void run() {
-                try {
-                    latch.countDown();
-                    queue.put("item");
-                } catch (InterruptedException e) {
-                    interrupted.set(true);
-                }
+            public void doRun() throws Throwable {
+                putLatch.countDown();
+                queue.put("item");
             }
         };
         thread.start();
 
-        latch.await(10, SECONDS);
-        thread.interrupt();
-        thread.join(5000);
+        assertOpenEventually(putLatch);
 
-        assertTrue("Expected queue.put() to be interrupted, but interrupted state is: " + interrupted, interrupted.get());
+        thread.interrupt();
+
+        thread.assertFailsEventually(InterruptedException.class);
     }
 
     private HazelcastInstance[] createHazelcastInstances() {

